cpp/src/arrow/engine/CMakeLists.txt

```diff
@@ -66,6 +66,8 @@ if(MSVC)
   list(APPEND SUBSTRAIT_SUPPRESSED_WARNINGS "/wd4251")
 endif()
 
+file(MAKE_DIRECTORY "${SUBSTRAIT_GEN_DIR}/substrait")
```

Comment:
I'm not sure, to be honest. See failing crossbox build above, which I wasn't able to reproduce. I figured it can't hurt to explicitly create the directory, so why not.

Reply:
CMake is sometimes finicky with non-existing directories, so I'm not surprised.